hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java

```diff
@@ -1502,13 +1502,6 @@ private int getWhenNoSSLDetected() {
     public static final HazelcastProperty SHUTDOWNHOOK_POLICY
             = new HazelcastProperty("hazelcast.shutdownhook.policy", "TERMINATE");
 
-    /**
-     * XML and system property for setting the hot restart required free space.
-     * By default, hot restart requires at least 15% free HD space.
-     */
-    public static final HazelcastProperty HOT_RESTART_FREE_NATIVE_MEMORY_PERCENTAGE
```

Comment:
Is this not a breaking change?

Reply:
Good point, not sure if this class is part of public api? If so, it is a breaking change. But AFAIK, only property names are mentioned in reference manual and no reference to this class. 
In any case, should we print a warning if the property is set? WDYT?

Reply:
If the class is in a package that contains `internal` or `impl`, it's private. Otherwise it's public, regardless of whether it's a mistake or not.

Sure, warning is fine, also deprecating it is good so we can perhaps clean it up in a future release.

Reply:
marked as deprecated